admin-dev/themes/new-theme/js/pages/monitoring/index.js

```diff
@@ -0,0 +1,80 @@
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+import Grid from '../../components/grid/grid';
+import FiltersResetExtension from '../../components/grid/extension/filters-reset-extension';
+import SortingExtension from '../../components/grid/extension/sorting-extension';
+import SubmitRowActionExtension
+  from '../../components/grid/extension/action/row/submit-row-action-extension';
+import LinkRowActionExtension from '../../components/grid/extension/link-row-action-extension';
+import DeleteCategoryRowActionExtension
+  from '../../components/grid/extension/action/row/category/delete-category-row-action-extension';
+import AsyncToggleColumnExtension
+  from '../../components/grid/extension/column/common/async-toggle-column-extension';
+import FiltersSubmitButtonEnablerExtension
+  from '../../components/grid/extension/filters-submit-button-enabler-extension';
+import ReloadListActionExtension from '../../components/grid/extension/reload-list-extension';
+import ExportToSqlManagerExtension
+  from '../../components/grid/extension/export-to-sql-manager-extension';
+import ColumnTogglingExtension from '../../components/grid/extension/column-toggling-extension';
+import SubmitBulkExtension from '../../components/grid/extension/submit-bulk-action-extension';
+import BulkActionCheckboxExtension
+  from '../../components/grid/extension/bulk-action-checkbox-extension';
+
+const $ = window.$;
+
+$(() => {
+  const emptyCategoriesGrid = new Grid('empty_category');
+
+  emptyCategoriesGrid.addExtension(new FiltersResetExtension());
+  emptyCategoriesGrid.addExtension(new SortingExtension());
+  emptyCategoriesGrid.addExtension(new ReloadListActionExtension());
+  emptyCategoriesGrid.addExtension(new SubmitRowActionExtension());
+  emptyCategoriesGrid.addExtension(new LinkRowActionExtension());
+  emptyCategoriesGrid.addExtension(new AsyncToggleColumnExtension());
+  emptyCategoriesGrid.addExtension(new DeleteCategoryRowActionExtension());
+  emptyCategoriesGrid.addExtension(new FiltersSubmitButtonEnablerExtension());
+
+  [
+    'product_with_combination',
```

Comment:
@matks, then `zero_quantity_product_with_combination`maybe?
And another list should also be like `zero_quantity_product_without_combination`?

Reply:
`zero_quantity_product_with_combinationmaybe` 👍 
For the factory name, at least put something in it like "Bad" : `BadProductWithoutCombinationGridDefinitionFactory` so that from the class name we understand what this is about